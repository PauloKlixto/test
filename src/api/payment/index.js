import request from '@/api/plugins/request'

export default {
  async fechPayment(body, success, error, done) {
    return await request(
      {
        method: 'post',
        url: 'pagar/',
        body
      },
      success,
      error,
      done
    )
  }
}
