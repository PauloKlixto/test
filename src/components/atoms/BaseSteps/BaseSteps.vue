<template>
  <section class="BaseSteps" :class="BaseStepsClasses">
    <section class="BaseSteps__list" v-for="item in steps" :key="item.id">
      <Icon v-show="item.active" class="BaseSteps__icon" name="check" />
      <span v-show="!item.active" class="BaseSteps__step">{{ item.step }}</span>
      <span class="BaseSteps__name">{{ item.name }}</span>
    </section>
    <section class="BaseSteps__responsive">
      <strong>Etapa {{ activeStep.step }}</strong> de {{ steps.length }}
    </section>
  </section>
</template>

<script>
export default {
  name: 'BaseSteps',
  props: {
    steps: {
      type: Array,
      required: true
    },
    invertColors: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    activeStep() {
      return this.steps.filter(step => step.active)[0]
    },
    BaseStepsClasses() {
      return { 'BaseSteps--inverted': this.invertColors }
    }
  }
}
</script>

<style lang="scss" scoped>
$responsiveValue: 960px;

.BaseSteps {
  font-size: 13px;
  color: $color-red;
  display: flex;

  &--inverted {
    color: $color-white;

    .BaseSteps__list {
      &:before {
        @include chevronAlt($color-white);
      }
    }

    .BaseSteps__icon {
      border: 1px solid $color-white;
    }

    .BaseSteps__step {
      border: 1px solid $color-white;
    }
  }

  &__responsive {
    display: none;

    @media (max-width: $responsiveValue) {
      display: block;
    }
  }

  &__list {
    position: relative;
    padding-left: 50px;
    display: flex;
    align-items: center;
    &:before {
      @include chevronAlt($color-red);
      left: 20px;
      transform: rotateZ(135deg) translateY(0px);
    }
    &:first-child {
      padding-left: 0;
      &:before {
        display: none;
      }
    }
    @media (max-width: $responsiveValue) {
      display: none;
    }
  }
  &__icon {
    display: flex;
    border: 1px solid $color-red;
    border-radius: 50%;
    width: 13px;
    height: 13px;
    padding: 4px;
    margin-right: 7.5px;
    background-color: $color-red;
  }
  &__step {
    display: flex;
    border: 1px solid $color-red;
    border-radius: 50%;
    width: 22px;
    height: 22px;
    align-items: center;
    justify-content: center;
    font-weight: bold;
    margin-right: 7.5px;
  }
  &__name {
    display: block;
  }
}
</style>
