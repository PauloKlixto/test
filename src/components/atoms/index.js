import BaseBackButton from './BaseBackButton/BaseBackButton'
import BaseButton from './BaseButton/BaseButton'
import BaseCreditCard from './BaseCreditCard/BaseCreditCard'
import BaseHeading from './BaseHeading/BaseHeading'
import BaseSelect from './BaseSelect/BaseSelect'
import BaseSteps from './BaseSteps/BaseSteps'

export {
  BaseBackButton,
  BaseButton,
  BaseCreditCard,
  BaseHeading,
  BaseSelect,
  BaseSteps
}
