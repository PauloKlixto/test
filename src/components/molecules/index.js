import InputGroup from './InputGroup/InputGroup'
import DynamicForm from './DynamicForm/DynamicForm'

export { InputGroup, DynamicForm }
