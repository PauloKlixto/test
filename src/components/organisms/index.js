import TheHeader from './TheHeader/TheHeader'

export { TheHeader }
