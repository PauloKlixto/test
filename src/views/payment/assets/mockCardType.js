export default {
  company: '',
  name: 'creditCardNumber',
  valid: false
}
