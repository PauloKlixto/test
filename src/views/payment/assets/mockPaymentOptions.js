export default [
  {
    id: 0,
    name: '12x R$1.000,00 sem juros'
  },
  {
    id: 1,
    name: '6x R$2.000,00 sem juros'
  },
  {
    id: 2,
    name: '3x R$4.000,00 sem juros'
  },
  {
    id: 3,
    name: '1x R$12.000,00 sem juros'
  }
]
