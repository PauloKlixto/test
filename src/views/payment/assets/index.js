import steps from './mockSteps'
import paymentOptions from './mockPaymentOptions'
import mockCardType from './mockCardType'

export { steps, paymentOptions, mockCardType }
