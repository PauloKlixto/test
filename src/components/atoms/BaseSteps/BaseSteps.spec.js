/* eslint-disable no-undef */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import BaseSteps from './BaseSteps'
import { steps } from '@/views/payment/assets'
import { Icon } from '@/components/photons'

const localVue = createLocalVue()
localVue.component('Icon', Icon)

describe('BaseSteps', () => {
  const WRAPPER = shallowMount(BaseSteps, {
    localVue,
    propsData: {
      steps
    }
  })

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Componente nÃ£o alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})
