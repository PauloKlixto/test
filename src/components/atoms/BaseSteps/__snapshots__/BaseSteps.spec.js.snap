// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSteps Componente não alterado 1`] = `
"<section class=\\"BaseSteps\\">
  <section class=\\"BaseSteps__list\\">
    <icon-stub name=\\"check\\" path=\\"icons\\" class=\\"BaseSteps__icon\\"></icon-stub> <span class=\\"BaseSteps__step\\" style=\\"display: none;\\">1</span> <span class=\\"BaseSteps__name\\">Carrinho</span>
  </section>
  <section class=\\"BaseSteps__list\\">
    <icon-stub name=\\"check\\" path=\\"icons\\" class=\\"BaseSteps__icon\\" style=\\"display: none;\\"></icon-stub> <span class=\\"BaseSteps__step\\">2</span> <span class=\\"BaseSteps__name\\">Pagamento</span>
  </section>
  <section class=\\"BaseSteps__list\\">
    <icon-stub name=\\"check\\" path=\\"icons\\" class=\\"BaseSteps__icon\\" style=\\"display: none;\\"></icon-stub> <span class=\\"BaseSteps__step\\">3</span> <span class=\\"BaseSteps__name\\">Confirmação</span>
  </section>
  <section class=\\"BaseSteps__responsive\\"><strong>Etapa 1</strong> de 3
  </section>
</section>"
`;
